File: in_Python/1047 Remove All Adjacent Duplicates In String.py
class Solution(object):
    def removeDuplicates(self, S):
        """
        :type S: str
        :rtype: str
        """
        st = []
        for i in list(S):
            if len(st) != 0 and st[-1] == i:
                st.pop()
            else:
                st.append(i)           
        res = ""
        for i in st: res += i
        return res
        

File: in_Java/1047 Remove All Adjacent Duplicates In String.java
class Solution {
    public String removeDuplicates(String S) {
        Stack<Character> st = new Stack<>();
        for(char c : S.toCharArray()){
            if(!st.isEmpty() && st.peek() == c){
                st.pop();
            }else{
                st.push(c);
            }
        }
        String res = "";
        while(!st.isEmpty()) res = st.pop().toString() + res;
        return res;
    }
}